shared/actions/config/index.js

```diff
@@ -112,13 +130,19 @@ const dispatchSetupEngineListeners = () => {
 }
 
 let _firstTimeConnecting = true
-const startHandshake = () => {
+const startHandshake = (state: TypedState) => {
   const firstTimeConnecting = _firstTimeConnecting
   _firstTimeConnecting = false
-  return Saga.put(ConfigGen.createDaemonHandshake({firstTimeConnecting}))
+  return Saga.put(
+    ConfigGen.createDaemonHandshake({firstTimeConnecting, version: state.config.daemonHandshakeVersion + 1})
```

Comment:
hmm...i think it should be the reducer that increments the version. Otherwise it's possible that multiple `createDaemonHandshake` actions would be in flight and yet the version would increment only once